### ISSUE TITLE
Validate URL Living Standard syntax in URL-typed attributes

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3848]
## Summary

The largest single residual on the nu-validator coverage benchmark is **1246 nu-only fixtures** that all report URL Living Standard parse errors:

- `tests/external/validator/tests/html/invalid-attr/**` — **1177 fixtures** (74% of the entire `invalid-attr` category)
- `tests/external/validator/tests/html/microdata/itemid/**` — **39 fixtures**
- `tests/external/validator/tests/html/microdata/itemtype/**` — **30 fixtures**

All three groups feed URL-typed attributes (`href`, `src`, `action`, `cite`, `data`, `formaction`, `manifest`, `poster`, `itemid`, `itemtype`, etc.). nu-validator runs them through a URL Living Standard validator and reports specific error codes. markuplint's `URL` type checker in `packages/@markuplint/types/src/whatwg/check-url.ts` (and the `invalid-attr` rule that consumes it) does not currently surface these errors.

## Confirmed nu-correct error categories (verbatim from audit log)

These are recorded in `.claude/skills/bench-triage/SKILL.md` as **"nu correct — NOT excluded"**, meaning the spec backs nu, and markuplint should report them too:

| nu message head | URL LS violation code |
| --- | --- |
| `URL includes credentials.` | [`invalid-credentials`](https://url.spec.whatwg.org/#invalid-credentials) |
| `Expected a slash ("/").` (special-scheme URL missing `//`) | [`special-scheme-missing-following-solidus`](https://url.spec.whatwg.org/#special-scheme-missing-following-solidus) |
| `Backslash ("\\") used as path segment delimiter.` | [`invalid-reverse-solidus`](https://url.spec.whatwg.org/#invalid-reverse-solidus) |
| `Illegal character in [path / fragment / domain / port].` | [`invalid-URL-unit`](https://url.spec.whatwg.org/#invalid-url-unit) (non-URL code points / malformed percent-encoding) |
| `Windows drive letter uses "\|" instead of ":"` | [`file-invalid-Windows-drive-letter`](https://url.spec.whatwg.org/#file-invalid-windows-drive-letter) / [`file-invalid-Windows-drive-letter-host`](https://url.spec.whatwg.org/#file-invalid-windows-drive-letter-host) |
| `Percentage ("%") is not followed by two hexadecimal digits.` | [`invalid-URL-unit`](https://url.spec.whatwg.org/#invalid-url-unit) (malformed percent-encoding) |
| `"file" scheme is not followed by "//".` | URL LS file-scheme parser (`file slash` / `file host` states) |

Sample fixture distribution from `microdata/itemid` (39 fixtures, but representative of `invalid-attr` patterns too):

```
12  Expected a slash ("/").                                   [special-scheme-missing-following-solidus]
11  URL includes credentials.                                  [invalid-credentials]
 6  "file" scheme is not followed by "//".                     [file-scheme-missing-following-solidus equivalent]
 4  Backslash ("\") used as path segment delimiter.            [invalid-reverse-solidus]
 2  Illegal character in fragment.                             [invalid-URL-unit]
 2  Illegal character in path segment.                         [invalid-URL-unit]
 1  Windows drive letter uses "|" instead of ":".              [file-invalid-Windows-drive-letter]
 1  Expected a token character or a semicolon.                 [itemtype-specific token grammar]
```

## Scope

The fix is a **single, focused implementation project**: extend `@markuplint/types`'s `URL` type checker to invoke a real URL Living Standard validator (or implement one inline). Once that lands, every URL-typed attribute consumed by `invalid-attr` (and the `attr-value` rule) inherits the new validation, flipping all 1246 fixtures in one shot.

The corpus already contains pinned regression fixtures, so verification is straightforward via `yarn bench:update --target markuplint && yarn bench:compare` (no Docker needed once nu raw snapshots exist).

## Suggested approach

1. Adopt or build a URL LS-conformant validator. Candidates:
   - [`whatwg-url`](https://github.com/jsdom/whatwg-url) — surfaces validation errors via the parser's reporting hooks (Node.js native `URL` swallows them silently).
   - Inline reimplementation following [URL LS §4 — URL parsing](https://url.spec.whatwg.org/#url-parsing).
2. Plumb the structured error signals through the `URL` type checker so callers can map them to `invalid-attr` violation messages.
3. Pre-flight against the bench: confirm no `match-clean` fixture regresses (URL LS strictness must not over-fire on legitimate URLs in the corpus).
4. Extend the existing unit coverage in `packages/@markuplint/types/src/whatwg/check-url.spec.ts` per audited error category.

## Cross-references

- bench-triage audit log: `.claude/skills/bench-triage/SKILL.md` (search for "URL LS")
- Existing URL checker: `packages/@markuplint/types/src/whatwg/check-url.ts`
- Existing URL checker tests: `packages/@markuplint/types/src/whatwg/check-url.spec.ts`
- URL type registry: `packages/@markuplint/types/src/defs.ts` (search for the `'URL'` and `'URLSet'` entries)
- Consumer rule: `packages/@markuplint/rules/src/invalid-attr/`

## Why this matters

This single Issue accounts for **~92% of the residual nu-only backlog** on the bench (1246 / 1345). Closing it brings overall nu-validator parity from ~57% to ~80%+. v5 stable's coverage story hinges on this work (see `tests/external/snapshots/diff/summary.md`).
